### ISSUE TITLE
Copy-paste from OneNote creates unwanted image files

### ISSUE THREAD

**Issue by cr-mocz** [jaap-karssenberg/zim-desktop-wiki#214]
Copying and pasting a regular section of text from OneNote into a Zim notebook results in a file called pasted_image.png being created and displayed inline.

The image displays fine, but it is not editable, so when the user notices this the logical next action is to undo the paste with Ctrl+Z or remove it with the backspace/delete key (and instead paste it into a notepad then into Zim so it will come in as text).

When one of those actions is taken (Undo or backspace), the image disappears from the notebook page, but remains in the file system and needs to be manually deleted.  If the image is not deleted manually and this process is repeated on the same notebook, the png files will accumulate with names pasted_image001.png and so on.

Occurs with Zim versions 0.65 through 0.67 using Windows 7 Home Premium SP1 (64-bit) and Microsoft Office Professional Plus 2013 (15.0.4953.1001).  The issue only seems to occur with OneNote - it does NOT occur when pasting from Word.

**Comment by Osndok** [comment]
Supposing they undo the backspace... or 'redo' the undo... how do we get the file back?

**Comment by cr-mocz** [comment]
Good question.  I guess if we define the bug as the fact that an image was created in the first place (rather than the fact that sticks around) then that is not a concern.  My expectation when I copy/paste text from OneNote into Zim is that it comes in as editable text, not an image.  Wouldn't you agree?

**Comment by Osndok** [comment]
Yes, and that part is a very strange bug indeed. It would help to know the exact browser/version you copied the text from, and the "--debug" output around the time of the paste (I think it says "clipboard offers X, Y, Z", or something).

**Comment by Osndok** [comment]
Sorry, I confused OneNote with EverNote... ignore the comment about the web browser.

**Comment by Osndok** [comment]
Apparently it is a known issue?
* http://www.onenotegem.com/document/copy-onenote-text-to-clipboard-as-plain-text-and-paste-to-other-applications-as-plain-text
* https://discussion.evernote.com/topic/96090-pasting-from-onenote-to-evernote-pastes-as-a-graphicimage/

**Comment by cr-mocz** [comment]
Looking at the Debug output, it appears that OneNote offers several format types for the paste and Zim chooses image/bmp

Below I pasted 2 separate items, the first one is from OneNote and has the bug, and the second one is from Notepad++ and pastes normally


DEBUG: Action: paste
DEBUG: **Targets available for paste: ('DataObject', 'OneNote 2010 Internal', 'HTML Format', 'UTF8_STRING', 'image/bmp', 'Ole Private Data')**, we want parsetree
DEBUG: Requesting data for image/bmp
DEBUG: Creating attachment dir: <zim.notebook.layout.FilesAttachmentFolder object at 0x0357C830>
DEBUG: Saving image from clipboard to C:\Zim\Active_scratchpad\pasted_image.png
DEBUG: Pasting link: file:///C:/Zim/Active_scratchpad/pasted_image.png (type: file, isimage: True)


DEBUG: Action: paste
DEBUG: Targets available for paste: ('UTF8_STRING',), we want parsetree
DEBUG: Requesting data for UTF8_STRING

**Comment by Osndok** [comment]
I think the fix is straight-forward, but I wonder if Jaap had a reason for preferring images?

If you like, you can try this (wholly untested but straight forward) change in the mean time:
* https://github.com/Osndok/zim-desktop-wiki/tree/help-214
* https://github.com/Osndok/zim-desktop-wiki/commit/4c6be346e01bf5352236ed0a43981bfad4a28f14

**Comment by cr-mocz** [comment]
That fix works for me!  I appreciate your time.  I am new to this code and it was very helpful for me to go through the debug and see where you made the fix.  I guess we will find out soon if there was a reason for preferring images, but either way, thank you!

**Comment by jaap-karssenberg** [comment]
Same issue can be seen with PowerPoint and other applications.

Reason I made image the default is that I have no way of knowing whether the clipboard contains text content only, or a complete drawing. E.g. pasting a powerpoint slide as text you loose a lot of content.

Only real solution is to allow "paste as" in zim and let the user choose between text and image content. Probably smart to remember the last choice as the default action.

**Comment by jaap-karssenberg** [comment]
P.S. not a good topic to start hacking on as the clipboard is completely broken in the Gtk3 port :disappointed: so code will have to revised anyway.

**Comment by jaap-karssenberg** [comment]
* Paste-As menu item to allow choice
* Preference for default choice
